src/plugins/discover/public/application/main/components/top_nav/open_alerts_popover.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useState, useMemo } from 'react';
+import ReactDOM from 'react-dom';
+import { I18nStart } from 'kibana/public';
+import { EuiWrappingPopover, EuiLink, EuiContextMenu } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ISearchSource } from '../../../../../../data/common';
+import { KibanaContextProvider, useKibana } from '../../../../../../kibana_react/public';
+import { DiscoverServices } from '../../../../build_services';
+import { updateSearchSource } from '../../utils/update_search_source';
+
+const container = document.createElement('div');
+let isOpen = false;
+
+interface AlertsPopoverProps {
+  onClose: () => void;
+  anchorElement: HTMLElement;
+  searchSource: ISearchSource;
+}
+
+export function AlertsPopover(props: AlertsPopoverProps) {
+  const dataView = props.searchSource.getField('index')!;
+  const searchSource = props.searchSource;
+  const { services } = useKibana<DiscoverServices>();
+  const [alertFlyoutVisible, setAlertFlyoutVisibility] = useState<boolean>(false);
+
+  const { triggersActionsUi } = useKibana<DiscoverServices>().services;
+
+  const onCloseAlertFlyout = useCallback(
+    () => setAlertFlyoutVisibility(false),
+    [setAlertFlyoutVisibility]
+  );
+
+  const getParams = useCallback(() => {
+    const nextSearchSource = searchSource.createCopy();
+    updateSearchSource(nextSearchSource, true, {
+      indexPattern: searchSource.getField('index')!,
+      services,
+      sort: [],
+      useNewFieldsApi: true,
+    });
+    const serializedFields = nextSearchSource.getSerializedFields();
+
+    return {
+      searchSource: serializedFields,
+    };
+  }, [searchSource, services]);
+
+  const AddAlertFlyout = useMemo(() => {
```

Comment:
nit: I found other teams named it like `{ruleType}AlertFlyout`, so seems like SearchThresholdAlertFlyout will be nice here. But it is on your vision and choice, I'm ok to have as it is.

Reply:
yes, makes sense, did that